### PR TITLE
Gen3: Use new Button

### PR DESCRIPTION
**Pull request by denysoblohin-okta**
## PR Checklist

- [x] Have you verified the basic functionality for this change?
- [ ] Did you add tests, as appropriate, following our [Automated Test guidelines](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/2676497890/Automated+Testing+in+the+Signin+Widget)?
- [ ] Did you follow our [Security Best Practices](https://oktawiki.atlassian.net/wiki/display/eng/Security+Best+practices)?
- [ ] Did you verify the change by running [downstream monolith artifact](https://oktawiki.atlassian.net/wiki/spaces/eng/pages/102897979/Sign-in+Widget+Development#Sign-inWidgetDevelopment-Instructionstocreateandrunthedownstreamartifact(d16t))? (Provide link to build below)
- [ ] Does this PR include noticeable changes to the UI? (If yes, attach screenshots/video below)

### Issue:

- [OKTA-678315](https://oktainc.atlassian.net/browse/OKTA-678315)

### CODE REVIEW

File: src/v3/src/util/theme.ts [hunk 1]
```diff
@@ -123,6 +123,16 @@ export const createThemeAndTokens = (
   // Merge default Odyssey 1.x theme with component overrides
   const themeOverride = mergeThemes(baseOdysseyTheme, {
     components: {
+      MuiButton: {
+        styleOverrides: {
+          root: {
+            display: 'flex',
+            justifyContent: 'center',
+            alignItems: 'center',
+            whiteSpace: 'normal',
```
Comment (trevoring-okta) [inline src/v3/src/util/theme.ts:132]: Added to fix text overflow bug, right?

File: src/v3/src/components/WebAuthNSubmitButton/WebAuthNSubmitButton.tsx [hunk 2]
```diff
@@ -89,15 +90,14 @@ const WebAuthNSubmit: UISchemaElementComponent<{
       justifyContent={showLoading ? 'center' : undefined}
     >
```
Comment (trevoring-okta) [inline src/v3/src/components/WebAuthNSubmitButton/WebAuthNSubmitButton.tsx:91]: Per latest discussion on auth list PR, can you convert all buttons' spacing values to use the closest approximate Odyssey spacing token

File: src/v3/src/components/LaunchAuthenticatorButton/LaunchAuthenticatorButton.tsx [hunk 3]
```diff
@@ -79,34 +79,19 @@ const LaunchAuthenticatorButton: UISchemaElementComponent<{
...
-        alignItems="center"
-        justifyContent="center"
-      >
-        <Box
-          sx={{
-            marginInlineEnd: '5px',
-            // keep the icon from stretching the button vertically
-            marginBlockEnd: '-3px',
-          }}
-        >
```
Comment (trevoring-okta) [inline src/v3/src/components/LaunchAuthenticatorButton/LaunchAuthenticatorButton.tsx:99]: Have you tested whether this fix was reintroduced by removing these styles?

File: src/v3/src/components/Button/Button.tsx [hunk 4]
```diff
@@ -79,24 +81,24 @@ const Button: UISchemaElementComponent<{
...
+      isFullWidth={wide ?? true}
+      buttonFocusRef={focusRef}
+      isDisabled={loading || disabled}
+      startIcon={loading ? (
+        <Spinner
+          color={!variant || variant === 'primary' ? tokens.HueNeutralWhite : undefined}
+        />
+      ) : ButtonImageIcon}
+      ariaDescribedBy={ariaDescribedBy}
+      testId={dataSe ?? dataType}
```
Comment (trevoring-okta) [inline src/v3/src/components/Button/Button.tsx:96]: Ideally Odyssey would give us some sort of `data-type` attribute but I guess there aren't any cases where we render buttons that need both `data-se` and `data-type`. We should definitely document this in the testing selector changes doc

File: src/v3/src/util/formUtils.ts [hunk 5]
```diff
@@ -77,8 +76,7 @@ export const getCustomButtonElements = (widgetProps: WidgetProps): ButtonElement
       options: {
         type: ButtonType.BUTTON,
         step: '',
-        dataSe: customButton.dataAttr,
-        classes: classNames(customButton.className, 'default-custom-button'),
```
Comment (trevoring-okta) [inline src/v3/src/util/formUtils.ts:81]: Can we now remove `classes` from the `ButtonElement` schema here? 
https://github.com/okta/okta-signin-widget/blob/a692da7a3705cd04e3af62bf469db193e3f0c07c/src/v3/src/types/schema.ts#L313

File: src/v3/src/util/theme.ts (hunk 1 above)
Comment (denysoblohin-okta) [reply src/v3/src/util/theme.ts:132]: Yes

File: src/v3/src/components/LaunchAuthenticatorButton/LaunchAuthenticatorButton.tsx (hunk 3 above)
Comment (denysoblohin-okta) [reply src/v3/src/components/LaunchAuthenticatorButton/LaunchAuthenticatorButton.tsx:99]: Yes. The fix is not needed after `styleOverrides` for `MuiButton`: https://github.com/okta/okta-signin-widget/pull/3502/files/7c133eb6bd7625bf0f82cbef3891281d3328ef3f#diff-18b2c20ca070dc0820e0e72efb48fb243e99865d2307d4b3f73353fd05debe10R126-R128

**Review by trevoring-okta — APPROVED** [review]
LGTM, just looks like you need to fix VRT and snapshot conflicts

**Comment by denysoblohin-okta** [comment]
Removed `color` from `Spinner` since it has been fixed in https://github.com/okta/okta-signin-widget/pull/3511

**Pull request merged**